lib/matplotlib/cm.py

```diff
@@ -83,8 +83,9 @@ def _generate_cmap(name, lutsize):
 
 # We silence warnings here to avoid raising the deprecation warning for
 # spectral/spectral_r when this module is imported.
-with _warnings.catch_warnings():
+with _warnings.catch_warnings(record=True):
```

Comment:
This seems way too complicated.  I would just invoke getitem using the superclass method, i.e. use `dict.__getitem__(cmap_d, name)` instead of `cmap_d[name]` at the right place.  (The first form does not raise a warning).